paddle/fluid/framework/parallel_executor.cc

```diff
@@ -165,9 +162,39 @@ void ParallelExecutor::Run(
     const std::unordered_map<std::string, LoDTensor> &feed_tensors) {
   platform::RecordBlock b(0);
   SplitTensorToPlaces(feed_tensors);
+
+  // Create local scopes.
+  for (auto &scope : member_->local_scopes_) {
+    Scope &local_scope = scope->NewScope();
+    *scope->Var(details::kLocalExecScopeName)->GetMutable<Scope *>() =
```

Comment:
What is usefule of `scope->Var(details::kLocalExecScopeName)` ? Please add comments.

Reply:
Currently, all non-persistable variables will be in the 'sub-scope of local scope'.